flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -1376,6 +1376,24 @@ class JobManager(
                     // addressed to the periodic checkpoint coordinator.
                     log.info("Received message for non-existing checkpoint " +
                       ackMessage.getCheckpointId)
+
+                    val classLoader = Option(libraryCacheManager.getClassLoader(jid)) match {
```

Comment:
Should we add comment here that we do this here to prevent lingering state if both coordinators already cleared the pending checkpoint?